Datatables 1972 - guidance on state and status (#2028)
grommet/hpe-design-system: aries-site/src/pages/components/datatable.mdx

<DataTable aria-describedBy="storage-pools-heading" />
```

### Status and State within a DataTable

Statuses within a DataTable should always have these accompanying elements:
- Shape
- Color

> - Content (***This is not restricted to just normal, warning, unknown, and critical)

Reviewer: Is this supposed to be bold?

Author: Nope, I'll remove the *** , good catch

More detailed information can be found on the [Status Indicator Design System Page](/templates/status-indicator#what-makes-up-a-status-indicator).

A State is the current stage of a process. A Status is the condition of the State. Status can also be thought of as a pointer to a State’s “health”. More detailed information can be found on the [Notification Design System Page](/components/notification#state-vs-status).

There are three separate scenarios in how Status and State can be displayed within a DataTable:

- [Separate columns for Status and State](#1-separate-columns-for-status-and-state)
- [Only Status column](#2-only-status-column)
- [Combined column for Status and State](#3-combined-column-for-status-and-state)

#### 1. Separate columns for Status and State

- Used when the Status column is not coupled with the displayed State column. (Ex. An equipment can have a “Running” State displayed within the DataTable, but have a “Normal” Status since it’s health is dependent on a State not shown within the DataTable.)
- Allows columns to be sorted independently from each other.

> - Clearly visually separates and labels Status and State to avoid false associations.

Reviewer: can we reword Im not sure Im following

<Box width="large">
  <Image src="/static/images/datatable-separate-columns-status-state.svg" alt="Example of a data table showing separate columns for status and state" />
</Box>

#### 2. Only Status column

- In some use cases, displaying Status on a DataTable is more important to users than State. If this is true for your use case, feel free to display only the Status in the DataTable.
- Used in situations where horizontal screen real-estate is an issue.
- If showing state is required, with this option State can be displayed once the user expands the cell for more information. (Ask yourself, “Does the user need to be able to see the State in the DataTable?”)

<Box width="large">
  <Image src="/static/images/datatable-only-status-column.svg" alt="Example showing a status column in the data table pointing to the state within the details" />
</Box>

#### 3. Combined column for Status and State

- Used when the Status and State fields are coupled with each other. Coupled Statuses are linked 1:1 to specific States and can be viewed as a single element.

> - (Note) The shape is represented as Status and content as State. (Ex. The column would show: Status Indicator Icon + State.)

Reviewer 2: ```suggestion
- A single column presents both status and state. Status represented by a [Status Indicator](/templates/status-indicator) and state as the accompanying label.
```

> - Status and State are sorted together rather than individually.

Reviewer 2: ```suggestion
- Status and state are sorted together rather than individually.
```

> - (Note) These are rarely used, be sure not to create false associations when Status and State are not coupled.

Reviewer 2: ```suggestion
- **Note**: A single column is rarely used. Ensure false associations are not created when status and state are not coupled.
```
